core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```
@@ -140,6 +142,73 @@ class KRaftMetadataCache(val brokerId: Int) extends MetadataCache with Logging w
+        val nextIndex = if (upperIndex < partitions.size()) upperIndex else -1
+        for (partitionId <- startIndex until upperIndex) {
+          topic.partitions().get(partitionId) match {
+            case partition : PartitionRegistration => {
+              val filteredReplicas = maybeFilterAliveReplicas(image, partition.replicas,
+                listenerName, false)
+              val filteredIsr = maybeFilterAliveReplicas(image, partition.isr, listenerName, false)
+              val offlineReplicas = getOfflineReplicas(image, partition, listenerName)
+              val maybeLeader = getAliveEndpoint(image, partition.leader, listenerName)
+              maybeLeader match {
+                case None =>
+                  result.append(new DescribeTopicPartitionsResponsePartition()
+                    .setPartitionIndex(partitionId)
+                    .setLeaderId(MetadataResponse.NO_LEADER_ID)
+                    .setLeaderEpoch(partition.leaderEpoch)
+                    .setReplicaNodes(filteredReplicas)
+                    .setIsrNodes(filteredIsr)
+                    .setOfflineReplicas(offlineReplicas)
+                    .setEligibleLeaderReplicas(Replicas.toList(partition.elr))
+                    .setLastKnownElr(Replicas.toList(partition.lastKnownElr)))
+                case Some(leader) =>
+                  result.append(new DescribeTopicPartitionsResponsePartition()
+                    .setPartitionIndex(partitionId)
+                    .setLeaderId(leader.id())
+                    .setLeaderEpoch(partition.leaderEpoch)
+                    .setReplicaNodes(filteredReplicas)
+                    .setIsrNodes(filteredIsr)
+                    .setOfflineReplicas(offlineReplicas)
+                    .setEligibleLeaderReplicas(Replicas.toList(partition.elr))
+                    .setLastKnownElr(Replicas.toList(partition.lastKnownElr)))
+              }
+            }
+            case _ =>
```

Should we throw an ISE here rather than silently continue? Maybe we could just log an error

---

Definitely should log an error.

---

Done.